server/cromwell_routes.py

```diff
@@ -0,0 +1,185 @@
+            timestamp = datetime.now().astimezone().isoformat()
+            metadata = json.dumps(
+                get_analysis_runner_metadata(
+                    timestamp=timestamp,
+                    dataset=dataset,
+                    user=email,
+                    access_level=access_level,
+                    repo=repo,
+                    commit=commit,
+                    script=wf,
+                    description=params['description'],
+                    output=output_dir,
+                    driver_image=DRIVER_IMAGE,
+                    cwd=cwd,
+                    mode='cromwell',
+                )
+            )
+
+            # Publish the metadata to Pub/Sub. Wait for the result before running the batch.
+            publisher.publish(PUBSUB_TOPIC, metadata.encode('utf-8')).result()
+
+            user_name = email.split('@')[0]
+            batch_name = f'{user_name} {repo}:{commit}/cromwell/{wf}'
+
+            batch = hb.Batch(backend=backend, name=batch_name)
+
+            job = batch.new_job(name='driver')
+            job = prepare_git_job(
+                job=job,
+                dataset=dataset,
+                access_level=access_level,
+                repo=repo,
+                commit=commit,
+                metadata_str=metadata,
+                output_dir=output_dir,
+            )
+            job.env('OUTPUT', output_dir)
+            if cwd:
+                job.command(f'cd {quote(cwd)}')
+
+            deps_path = None
+            if libs:
+                deps_path = 'tools.zip'
+                job.command(
+                    'zip -r tools.zip ' + ' '.join(quote(s + '/') for s in libs)
+                )
+
+            cromwell_post_url = make_cromwell_url('api/workflows/v1/')
+            workflow_options = {
+                'user_service_account_json': service_account,
+                'google_project': project,
+                'jes_gcs_root': intermediate_dir,
+                'final_workflow_outputs_dir': output_dir,
+            }
+
+            workflow_inputs = None
+
+            job.command(
+                f"""
+cat '{json.dumps(workflow_options)}' > workflow_options.json
+curl -X POST "{cromwell_post_url}" \\
+    -H "accept: application/json" \\
+    -H "Content-Type: multipart/form-data" \\
+    -F "workflowSource=@{wf}" \\
+    {f'-F "workflowInputs=@{workflow_inputs}"' if workflow_inputs else ''} \\
+    -F "workflowOptions=@workflow-options.json;type=application/json" \\
+    {f'-F "workflowOptions=@{deps_path}"' if deps_path else ''} \\
+"""
+            )
+
+            url = run_batch_job_and_print_url(batch, wait=params.get('wait', False))
+
+            return web.Response(text=f'{url}\n')
+        except KeyError as e:
+            logging.error(e)
+            raise web.HTTPBadRequest(reason='Missing request parameter: ' + e.args[0])
+
+    @routes.get('/cromwell/engine/version')
```

Comment:
I think all of these we could just allow directly, especially if the non-destructive ones.

Reply:
Yep, sounds great!